homeassistant/components/go2rtc/__init__.py

```diff
@@ -54,38 +65,75 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         entry.async_on_unload(server.stop)
         server.start()
 
-    client = Go2RtcClient(async_get_clientsession(hass), entry.data[CONF_HOST])
-
-    provider = WebRTCProvider(client)
+    provider = WebRTCProvider(hass, entry.data[CONF_HOST])
     entry.async_on_unload(async_register_webrtc_provider(hass, provider))
     return True
 
 
 class WebRTCProvider(CameraWebRTCProvider):
     """WebRTC provider."""
 
-    def __init__(self, client: Go2RtcClient) -> None:
+    def __init__(self, hass: HomeAssistant,url:str,) -> None:
         """Initialize the WebRTC provider."""
-        self._client = client
+        self._hass = hass
+        self._url = url
+        self._session = async_get_clientsession(hass)
+        self._rest_client = Go2RtcRestClient(self._session, url)
+        self._sessions: dict[str, Go2RtcWsClient] = {}
 
     async def async_is_supported(self, stream_source: str) -> bool:
         """Return if this provider is supports the Camera as source."""
         return stream_source.partition(":")[0] in _SUPPORTED_STREAMS
 
     async def async_handle_web_rtc_offer(
-        self, camera: Camera, offer_sdp: str
-    ) -> str | None:
-        """Handle the WebRTC offer and return an answer."""
-        streams = await self._client.streams.list()
+        self,
+        camera: Camera,
+        offer_sdp: str,
+        session_id: str,
+        send_message: WebRTCSendMessage,
+    ) -> bool:
+        """Handle the WebRTC offer and return the answer via the provided callback.
+
+        Return value determines if the offer was handled successfully.
```

Comment:
From reading the code, the return value determines if the offer was successfully forwarded to Go2RtcWsClient. It can still return `True` even if Go2RtcWSWebClient rejects the offer.

Reply:
I think it would be useful to clarify how these should be used either here or in the developer documentation:
- boolean return value
- HomeAssisantError
- async error reporting

I would also think it makes sense to make the semantic of the provider similar to the camera entity handling method. (e.g. don't have a return value on one and not the other)